### PR TITLE
Add a prune method that can be used to remove updates that will have …

### PR DESCRIPTION
**Pull request by baldersheim**
…no effect when applied.

Currently will only affect ASSIGN updates.
@vekterli or @bratseth PR

### CODE REVIEW

File: document/src/main/java/com/yahoo/document/DocumentUpdate.java [hunk 1]
```diff
@@ -116,6 +119,35 @@ public DocumentUpdate applyTo(Document doc) {
         return this;
     }
 
+    /**
+     * Will prune away any field update that will not modify any field in the document.
```
Comment (bratseth) [inline document/src/main/java/com/yahoo/document/DocumentUpdate.java]: Drop "Will"

File: document/src/main/java/com/yahoo/document/DocumentUpdate.java [hunk 2]
```diff
@@ -116,6 +119,35 @@ public DocumentUpdate applyTo(Document doc) {
...
+     * @param doc
```
Comment (bratseth) [inline document/src/main/java/com/yahoo/document/DocumentUpdate.java]: Drop this line if you're not going to add doc.

File: document/src/main/java/com/yahoo/document/DocumentUpdate.java [hunk 3]
```diff
@@ -116,6 +119,35 @@ public DocumentUpdate applyTo(Document doc) {
...
+     * @return a reference to itself
+     * @throws IllegalArgumentException if the document does not have the same document type as this update
+     */
+    public DocumentUpdate prune(Document doc) {
+        if ( ! fieldPathUpdates.isEmpty()) return this;
+
+        if (!documentType.equals(doc.getDataType())) {
+            throw new IllegalArgumentException(
+                    "Document " + doc + " must have same type as update, which is type " + documentType);
```
Comment (bratseth) [inline document/src/main/java/com/yahoo/document/DocumentUpdate.java]: Output also the attempted type?

**Review by bratseth — APPROVED** [review]
LGTM, some nitpicking comments

File: document/src/main/java/com/yahoo/document/DocumentUpdate.java (hunk 1 above)
Comment (baldersheim) [reply document/src/main/java/com/yahoo/document/DocumentUpdate.java]: Fixed

File: document/src/main/java/com/yahoo/document/DocumentUpdate.java (hunk 2 above)
Comment (baldersheim) [reply document/src/main/java/com/yahoo/document/DocumentUpdate.java]: Fixed

File: document/src/main/java/com/yahoo/document/DocumentUpdate.java (hunk 3 above)
Comment (baldersheim) [reply document/src/main/java/com/yahoo/document/DocumentUpdate.java]: Fixed and deduplicated.

**Review by vekterli — APPROVED** [review]
:+1: Looks good, just one question on field path updates

**Pull request merged**